packages/nameguard-sdk/src/index.ts

```diff
@@ -462,16 +462,30 @@ export interface SecurePrimaryNameResult {
   nameguard_result: NameGuardReport | null;
 }
 
-// TODO: Let's apply more formalization to this error class.
-class NameGuardError extends Error {
+export enum NameGuardErrorType {
```

Comment:
Can we please document at least 1 sentence for each of these different error types to explain in more words what they represent?

Reply:
Added a doc block for the whole class and constructor now @lightwalker-eth 👍🏻